### ISSUE TITLE
Time Crisis 4 (Standalone or Arcade Ver.) - Framerate drops when running at >=60hz

### ISSUE THREAD

**Issue by SeongGino** [RPCS3/rpcs3#13822]
### Quick summary

Time Crisis 4 gameplay runs at sub-optimal 3D framerates but reports full speed when frame limiter is set to 60 or above, or VBlank frequency is set higher than 59hz.

### Details

Time Crisis 4, either the original disc release [BLUS30093] or the "Arcade Ver." as part of TIME CRISIS: RAZING STORM [BLUS30528], seems to perform very poorly on otherwise capable hardware when RPCS3 is set to run the game at 60hz or above. Gameplay seems to run at a (jerky) 30 fps, introducing dropped button inputs (i.e. the trigger button on either normal controllers or emulated Motion Controller), while the cursor moves smoothly at 60fps.

[Visual demonstration of the issue.](https://puu.sh/JG9ud/7503193883.mp4) Notice how the shadows flicker like this, and the cursor moving smoothly while the camera jerkily pans down the staircase.

From what I can tell, either version of TC4 specifically requires forcing the game to run at **59hz** in the Advanced tab -> VBlank Frequency override - even the NTSC Fixup checkbox allegedly meant to fix timing for Space Channel 5 pt2 doesn't seem to be enough. While this fixes the problem and makes gameplay and input smoother, the performance still isn't optimal as frame pacing won't match either NTSC standard or most 60hz monitors.

With regards to the rest of TIME CRISIS: RAZING STORM, this does *not* seem to affect Deadstorm Pirates or Razing Storm (the namesake game itself); both games run properly at a flat 60FPS, even when RPCS3's limiter is off. [This video of TC4 through RAZING STORM on RPCS3](https://www.youtube.com/watch?v=kDKnETEdP9Y) seems to corroborate this, as the GPU config shown sets a 50fps cap - which would only be correct if I was running a PAL system, I suppose.

Only other issue that seems to bring this up is #4931, which somewhat handwaves the problem and is also several years old. If anything, this is just to clear the air on this game for anyone who's searching.

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11413586/RPCS3.log.gz)

### System configuration

CPU: AMD Ryzen 5 5600X
GPU: NVIDIA RTX 3060 Ti
OS: Arch Linux
Driver: NVIDIA (Proprietary) 530.41.03

**Comment by legend800** [comment]
I can confirm all of this (issue and workaround). I thought running at 30fps was the only fix to the input dropping. Thanks for detailing everything out.

**Comment by kd-11** [comment]
Try setting the frame limit to "ps3" mode in conjunction with the ntsc fixup.

**Comment by legend800** [comment]
For me, here's my results:

Framerate=auto, Vblank=59, NTSC=fix up on= broken input
Framerate=auto, Vblank=59, NTSC=fix up off = broken input
Framerate=ps3 native, Vblank=59, NTSC=fix up on = broken input
Framerate=50, Vblank=60, NTSC=fix up off = freeze on opening cinematic (2/2)

Framerate=50, Vblank=59, NTSC=fix up off = perfect
Framerate=30, Vblank=59, NTSC=fix up off = perfect

**Comment by SeongGino** [comment]
> Try setting the frame limit to "ps3" mode in conjunction with the ntsc fixup.

Setting frame limit to *PS3 Native*, VBlank reset back to 60 with NTSC Fixup checked, and it's the same issue. According to the FPS counter in the window title, it does seem to hover more closely around 59.94, but the game framerate (not cursor or reported FPS) is still bad and shadows are flickering like the other attempts at running beyond 59hz. Same with normal Auto frame limiter.

**Comment by dante3732** [comment]
Game work fine for me only if I using 50 FPS cap.

**Comment by dante3732** [comment]
> Also, @legend800 not sure why you set a frame limiter on in the latter two tests; you can set the frame limiter to _off_, it just requires the VBlank frequency only be set to 59hz to 'fix' the performance.

Yep, 59Hz fix issue.

**Comment by legend800** [comment]
Nope that doesn't fix the problem. framelimit/vsync off=off, Vblank frequency=59hz still losing bullets, even on a high end machine. Simply start game and start unloading clips fast = every few bullets will not fire. 

Framerate of 50 or 30 fps is the only thing that gives me perfect shots.

**Comment by SeongGino** [comment]
> Nope that doesn't fix the problem. framelimit/vsync off=off, Vblank frequency=59hz still losing bullets, even on a high end machine. Simply start game and start unloading clips fast = every few bullets will not fire.
> 
> Framerate of 50 (with 59hz) or 30 fps (vblank doesn't matter) is the only thing that gives me perfect shots.

Yeah, I can't reproduce that part at all. No missed shots at just 59hz. And I was mashing the shit out of the trigger/mouse button.

Either that's a setup thing, or a Windows vs. Linux difference?

**Comment by digant73** [comment]
is that issue still present?

**Comment by pepodmc3** [comment]
> is that issue still present?

yeah

**Comment by SeongGino** [comment]
Belated response, but definitely still an issue here with the latest version (as of today, 55aae4dd) unfortunately.

**Comment by Bag-Head** [comment]
Want to confirm on my system this is still the case with standalone TC4 and Razing Storm's version and I can only resolve it by forcing the frame limit to 50fps, anything else (even with VBlank set to 59Hz) doesn't fix the performance or input issues
